src/main/java/tradeUseCase/tradeInteractor.java

```diff
@@ -0,0 +1,96 @@
+package tradeUseCase;
+import MainEntities.CampaignAccess;
+import MainEntities.Player;
+import Properties.NormalProperty;
+import Properties.Property;
+
+
+import java.util.ArrayList;
+
+/**
+ * TradeInteractor is a class that handles the trade between two players.
+ */
+public class tradeInteractor implements tradeInputBoundary {
+    final tradeOutputBoundary output;
+    final CampaignAccess campaign;
+    Player player1;
+    Player player2;
+
+    ArrayList<NormalProperty> player1Properties;
+    ArrayList<NormalProperty> player2Properties;
+    int player1Cash;
+    int player2Cash;
+
+    public tradeInteractor(tradeOutputBoundary output, Player player2, CampaignAccess campaign, ArrayList<NormalProperty> player1Properties,
+                           ArrayList<NormalProperty> player2Properties, int player1Cash, int player2Cash) {
+        this.output = output;
+        this.player2 = player2;
+        this.campaign = campaign;
+        this.player1Properties = player1Properties;
+        this.player2Properties = player2Properties;
+        this.player1Cash = player1Cash;
+        this.player2Cash = player2Cash;
+    }
+
+    /**
+     * @param player1Properties The properties that the player who is trading is offering.
+     * @param player2Properties The properties that the player who is trading is asking for.
+     * @param player1Cash The amount of cash that the player who is trading is offering.
+     * @param player2Cash The amount of cash that the player who is trading is asking for.
+     */
+    public void trade(ArrayList<NormalProperty> player1Properties, ArrayList<NormalProperty> player2Properties,
+                      int player1Cash, int player2Cash) {
+        player1.setCash(player1.getCash() - player1Cash + player2Cash);
```

Comment:
You can also use the loseCash(amount) and gainCash(amount) method. i.e. player1.lostCash(amount), but setCash works as well